spras/containers.py

```diff
@@ -148,6 +150,43 @@ def run_container(framework: str, container_suffix: str, command: List[str], vol
     else:
         raise ValueError(f'{framework} is not a recognized container framework. Choose "docker" or "singularity".')
 
+def run_container_and_log(name: str, framework: str, container_suffix: str, command: List[str], volumes: List[Tuple[PurePath, PurePath]], working_dir: str, environment: str = 'SPRAS=True'):
+    """
+    Runs a command in the container using Singularity or Docker with associated pretty printed messages.
+    @param framework: singularity or docker
+    @param container_suffix: name of the DockerHub container without the 'docker://' prefix
+    @param command: command to run in the container
+    @param volumes: a list of volumes to mount where each item is a (source, destination) tuple
+    @param working_dir: the working directory in the container
+    @param environment: environment variables to set in the container
+    @return: output from Singularity execute or Docker run
+    """
+    print('Running {} on container framework "{}" with command: {}'.format(name, framework, ' '.join(command)), flush=True)
+    try:
+        out = run_container(framework=framework, container_suffix=container_suffix, command=command, volumes=volumes, working_dir=working_dir, environment=environment)
+        if out is not None:
+            if isinstance(out, list):
+                out = ''.join(out)
+            elif isinstance(out, dict):
+                if 'message' in out:
+                    # This is the format of a singularity message.
+                    # See https://singularityhub.github.io/singularity-cli/api/source/spython.main.html?highlight=execute#spython.main.execute.execute.
+                    if 'return_code' in out and not out['return_code'] == 0:
+                        print(f"(Program exited with non-zero exit code '{out['return_code']}')")
+                    out = ''.join(out['message'])
+                else:
+                    print("Note: This is an unknown message format - if you want this pretty printed, please file out an issue at https://github.com/Reed-CompBio/spras/issues/new.")
+                    out = str(out)
+            elif not isinstance(out, str):
+                out = str(out, "utf-8")
+            print(indent(out))
+    except docker.errors.ContainerError as err:
```

Comment:
`errors` reports `Cannot find reference 'errors' in '__init__.py' ` in my IDE. Do you see that as well?

Reply:
I don't see that.

Reply:
There isn't a stack trace, it is a static warning from PyCharm:
![image](https://github.com/user-attachments/assets/55183412-ce6f-4003-aaaf-2576a53e7a29)

If you don't see it, I'm not concerned.